### ISSUE TITLE
Use sentinel contract (metering) when filling tests

### ISSUE THREAD

**Issue by axic** [ewasm/tests#110]

**Comment by lrettig** [comment]
Do we want to refill all tests using sentinel contract metering?

**Comment by axic** [comment]
Yes, otherwise each of the VMs/clients need to implement the same logic, but we were postponing this until the code/specification for Sentinel settles. I'd hold back on this a bit more.
